projects/android-ms_sec_hackathon_2023/app/src/main/java/com/sample/hackathon/declaredaccessandroid/graph/GraphServiceFactory.java

```diff
@@ -17,13 +19,34 @@ public class GraphServiceFactory {
 
     private static final String BASE_URL_GRAPH = "https://graph.microsoft.com/v1.0/";
 
+    // TODO make blocking and use Future
```

Comment:
As mentioned in Teams chat -- I'll not mess with this for now since it seems to be working for you